blockchain/signers/signers.go

```diff
@@ -60,16 +72,50 @@ func Path(s *Signer, ks keySpace, itemIndexes ...uint64) [][]byte {
 	return path
 }
 
+// getBip0044Path returns the complete path for bip0044 derived keys
+func getBip0044Path(accountIndex uint64, change bool, addrIndex uint64) [][]byte {
+	var path [][]byte
+	path = append(path, BIP44Purpose[:]) //purpose
+	path = append(path, BTMCoinType[:])  //coin type
+	accIdxBytes := make([]byte, 8)
+	binary.LittleEndian.PutUint64(accIdxBytes, uint64(accountIndex))
+	path = append(path, accIdxBytes) //account index
+	branchBytes := make([]byte, 8)
+	if change {
+		binary.LittleEndian.PutUint64(branchBytes, uint64(1))
+	} else {
+		binary.LittleEndian.PutUint64(branchBytes, uint64(0))
+	}
+	path = append(path, branchBytes) //change
+	addrIdxBytes := make([]byte, 8)
+	binary.LittleEndian.PutUint64(addrIdxBytes[:], addrIndex)
+	path = append(path, addrIdxBytes[:]) //address index
+	return path
+}
+
+// Path returns the complete path for derived keys
+func Path(s *Signer, ks keySpace, change bool, addrIndex uint64) ([][]byte, error) {
+	switch s.DeriveRule {
+	case BIP0032:
+		return GetBip0032Path(s, ks, addrIndex), nil
+	case BIP0044:
+		return getBip0044Path(s.KeyIndex, change, addrIndex), nil
+	}
+	return nil, ErrDeriveRule
+}
+
 // Create creates and stores a Signer in the database
-func Create(signerType string, xpubs []chainkd.XPub, quorum int, keyIndex uint64) (*Signer, error) {
+func Create(signerType string, xpubs []chainkd.XPub, quorum int, keyIndex uint64, deriveRule uint8) (*Signer, error) {
 	if len(xpubs) == 0 {
 		return nil, errors.Wrap(ErrNoXPubs)
 	}
 
-	sort.Sort(sortKeys(xpubs)) // this transforms the input slice
-	for i := 1; i < len(xpubs); i++ {
-		if bytes.Equal(xpubs[i][:], xpubs[i-1][:]) {
-			return nil, errors.WithDetailf(ErrDupeXPub, "duplicated key=%x", xpubs[i])
+	xpubsMap := map[chainkd.XPub]byte{}
+	for _, xpub := range xpubs {
```

Comment:
for _, xpub := range xpubs {
    if _, ok := xpubsMap[xpub]{
        return nil, errors.WithDetailf(ErrDupeXPub, "duplicated key=%x", xpub)
    }
    xpubsMap[xpub] = 0
}